instructors.html

```diff
@@ -27,21 +27,45 @@
+<li><p>Give learners a five-minute overview of what version control does for them before diving into the watch-and-do practicals. Most of them will have tried to co-author papers by emailing files back and forth, or will have biked into the office only to realize that the USB key with last night's work is still on the kitchen table. Instructors can also make jokes about directories with names like &quot;final version&quot;, &quot;final version revised&quot;, &quot;final version with reviewer three's corrections&quot;, &quot;really final version&quot;, and, &quot;come on this really has to be the last version&quot; to motivate version control as a better way to collaborate and as a better way to back work up.</p></li>
+<li><p>Version control is typically taught after the shell, so collect learners' names during that session and create a repository for them to share with their names as both their IDs and their passwords.</p></li>
+<li><p>If your learners are advanced enough to be comfortable SSH, tell them they can use keys to authenticate on GitHub instead of passwords, but don't try to set this up during class: it takes too long, and is a distraction from the core ideas of the lesson.</p></li>
+<li><p>Be very clear what files learners are to edit and what user IDs they are to use when giving instructions: it is common for them to edit the file the instructor is working on rather than their own, or to use the instructor's user ID and password when committing.</p></li>
+<li><p>Be equally clear <em>when</em> they are to edit things: it's also common for someone to edit the file the instructor is editing and commit changes while the instructor is explaining what's going on, so that a conflict occurs when the instructor comes to commit the file.</p></li>
+<li><p>If some learners are using Windows, there will inevitably be issues merging files with different line endings. (Even if everyone's on some flavor of Unix, different editors may or may not add a newline to the last line of a file.) Take a moment to explain these issues, since learners will almost certainly trip over them again.</p></li>
+<li><p>We don't use a Git GUI in these notes because we haven't found one that installs easily and runs reliably on the three major operating systems, and because we want learners to understand what commands are being run. That said, instructors should demo a GUI on their desktop at some point during this lesson and point learners at <a href="http://git-scm.com/downloads/guis">this page</a>.</p></li>
+<li><p>Instructors should also show learners graphical diff/merge tools like <a href="https://sourcegear.com/diffmerge/">DiffMerge</a>.</p></li>
+<li><p>When appropriate, explain that we teach Git rather than CVS, Subversion, or Mercurial primarily because of GitHub's growing popularity: CVS and Subversion are now seen as legacy systems, and Mercurial isn't nearly as widely used in the sciences right now.</p></li>
```

Comment:
It would be better to mention that Git is faster and a better collaboration tool than CVS or Subversion. Popular services such as GitHub, Bitbucket or GitLab also make using Git an excellent tool choice. Mercurial, though similar to Git, has a smaller user base and many of the Git concepts apply if a user has a future need for Mercurial.